Data/ByteString/Internal.hs

```diff
@@ -648,6 +649,25 @@ concat = \bss0 -> goLen0 bss0 bss0
    concat [x] = x
  #-}
 
+-- | Repeats given ByteString n times. More efficient than default definition.
+times :: Integral a => a -> ByteString -> ByteString
+times n (BS fp len)
+  | n < 0 = error "stimes: non-negative multiplier expected"
+  | n == 0 = BS nullForeignPtr 0 -- definition of empty in Data.ByteString
```

Comment:
Can't we simply move `empty` to this module?

Reply:
Just looked through the module and noticed mempty is already used - I just did that.